pkg/ansible/controller/reconcile.go

```diff
@@ -84,9 +86,8 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 		duration, err := time.ParseDuration(ds)
 		if err != nil {
 			// Should attempt to update to a failed condition
-			if errmark := r.markError(u, request.NamespacedName, fmt.Sprintf("Unable to parse reconcile period annotation: %v", err)); errmark != nil {
-				logger.Error(errmark, "Unable to mark error annotation")
-			}
+			merr := r.markError(u, request.NamespacedName, fmt.Sprintf("Unable to parse reconcile period annotation: %v", err))
+			checkAPIErr(merr, logger)
```

Comment:
It will return a bool, however, its return has not been checked is correct? 
Should not it be checked and then, re-trigger the reconcile?

Reply:
The boolean value is not checked in most places to maintain consistency with current logic. However if `Reconcile` is changed in the future, `checkAPIErr`'s return value may be useful for logging errors.

Reply:
Why the error is not checked and the reconcile re-trigged here as it is done [here](https://github.com/operator-framework/operator-sdk/pull/1047/files#diff-9649888b3df1961dad649345545de4c0R129)?

Reply:
I'm pretty sure we need to check the not found error again because we try to retrieve the CR (again) after the playbook runs, since the playbook could have changes the CR or its status.

Reply:
Hi @joelanford,

See that this new func `checkAPIErr` was added in the lines: [90](https://github.com/operator-framework/operator-sdk/blob/5f8a40338dacbac3d9b9cea562f475f0b620f99d/pkg/ansible/controller/reconcile.go#L90), [129](https://github.com/operator-framework/operator-sdk/blob/5f8a40338dacbac3d9b9cea562f475f0b620f99d/pkg/ansible/controller/reconcile.go#L129), [144](https://github.com/operator-framework/operator-sdk/blob/5f8a40338dacbac3d9b9cea562f475f0b620f99d/pkg/ansible/controller/reconcile.go#L144), [156](https://github.com/operator-framework/operator-sdk/blob/5f8a40338dacbac3d9b9cea562f475f0b620f99d/pkg/ansible/controller/reconcile.go#L156), [127](https://github.com/operator-framework/operator-sdk/blob/5f8a40338dacbac3d9b9cea562f475f0b620f99d/pkg/ansible/controller/reconcile.go#L227) to replace:

```
logger.Error(errmark, "Unable to mark <specific descrition>")
```
So, does it as it has been done before this change:

// perform operation
// the operation has an error
// log a specific msg to allow troubleshooting and find out where the error was faced
// return `reconcileResult, err`

Will re-trigger the reconcile where has the check [apierrors.IsNotFound(err)](https://github.com/operator-framework/operator-sdk/blob/5f8a40338dacbac3d9b9cea562f475f0b620f99d/pkg/ansible/controller/reconcile.go#L70) to verify if the CR was deleted and log this info as stop the reconcile. So, it will always be checked and then, because of this, I do not see the point to add again the check in`checkAPIErr.`

IHMO:
- the previous code was better because for each scenario a specific log was added which allows us to know where exacly the error was faced. The generic msg in the logs `"Error while updating CR status"` does not help to troubleshoot it. 
- The check to verify if the CR was deleted is redundant at this point. It will be checked after any error faced in the reconcile since it should be retrigged as it has been done so far.
- The func returns a bool and it has not always been used which is very confusing.
- The func name is not making intuitive the understanding about what it does and when should be used.

Reply:
> It is out of the scope of this PR which is: Ansible and Helm operator reconcilers use a cached client for reads instead of the default unstructured client.

This is a good point. @estroz is there a reason we need to make these other changes or is there a lighter touch change that can be made?